src/openforms/submissions/tests/test_post_submission_event.py

```diff
@@ -0,0 +1,930 @@
+from unittest.mock import patch
+
+from django.core import mail
+from django.test import TestCase, override_settings
+from django.utils.translation import gettext_lazy as _
+
+from privates.test import temp_private_root
+
+from openforms.appointments.exceptions import AppointmentRegistrationFailed
+from openforms.appointments.tests.utils import setup_jcc
+from openforms.authentication.constants import AuthAttribute
+from openforms.emails.tests.factories import ConfirmationEmailTemplateFactory
+from openforms.forms.tests.factories import FormDefinitionFactory
+from openforms.payments.constants import PaymentStatus
+from openforms.payments.tests.factories import SubmissionPaymentFactory
+
+from ..constants import RegistrationStatuses
+from ..models import SubmissionReport
+from ..tasks import on_post_submission_event
+from .factories import SubmissionFactory
+
+
+@temp_private_root()
+@override_settings(CELERY_TASK_ALWAYS_EAGER=True, LANGUAGE_CODE="en")
+class TaskOrchestrationPostSubmissionEventTests(TestCase):
+    def test_submission_completed_cosign_and_payment_not_needed(self):
+        # The registration should happen since we are not waiting on payment/cosign
+        submission = SubmissionFactory.from_components(
+            components_list=[
+                {
+                    "key": "email",
+                    "type": "email",
+                    "label": "Email",
+                    "confirmationRecipient": True,
+                }
+            ],
+            form__name="Pretty Form",
+            submitted_data={"email": "test@test.nl"},
+            completed_not_preregistered=True,
+            cosign_complete=False,
+            confirmation_email_sent=False,
+            form__registration_backend="email",
+            form__registration_backend_options={"to_emails": ["test@registration.nl"]},
+            auth_info__attribute=AuthAttribute.bsn,
+            auth_info__value="111222333",
+            language_code="en",
+        )
+        ConfirmationEmailTemplateFactory.create(
+            form=submission.form,
+            subject="Confirmation of your {{ form_name }} submission",
+            content="Custom content {% appointment_information %} {% payment_information %} {% cosign_information %}",
+        )
+
+        with patch(
+            "openforms.registrations.contrib.email.plugin.EmailRegistration.register_submission"
+        ) as mock_registration, patch(
+            "openforms.submissions.public_references.get_reference_for_submission",
+            return_value="OF-TEST!",
+        ), patch(
+            "openforms.registrations.contrib.email.plugin.EmailRegistration.update_payment_status"
+        ) as mock_payment_status_update:
+            on_post_submission_event(submission.id)
```

Comment:
Trick from Chris to force formatting of context managers:

```py
with (
  cm1() as cv1,
  cm2() as cv2,
  ...,
):
```

leads to more readable code